### PR TITLE
[matplotplusplus,nodesoup] update to v1.2.0,add new port

### LINKED ISSUE #31724: [matplotplusplus] update to v1.2.0
### Library name

matplotplusplus

### New version number

1.2.0

### Other information that may be useful (release notes, etc...)

https://github.com/alandefreitas/matplotplusplus/releases/tag/v1.2.0

### PR DESCRIPTION
**Pull request by jiayuehua**
Fixes #31724

- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [X] SHA512s are updated for each updated download
- [X] ~The "supports" clause reflects platforms that may be fixed by this new version~
- [X] ~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by MonicaLiu0311** [comment]
The following error occurred when testing the feature on `x86-windows` and `x64-windows-static`:
```
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:112 (message):
    Command failed: "C:/Program Files/Microsoft Visual Studio/2022/Enterprise/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" -v
    Working Directory: E:/matplotplusplus/buildtrees/matplotplusplus/x86-windows-rel/vcpkg-parallel-configure
    Error code: 1
    See logs for more information:
      E:\matplotplusplus\buildtrees\matplotplusplus\config-x86-windows-dbg-CMakeCache.txt.log
      E:\matplotplusplus\buildtrees\matplotplusplus\config-x86-windows-rel-CMakeCache.txt.log
      E:\matplotplusplus\buildtrees\matplotplusplus\config-x86-windows-out.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_configure.cmake:248 (vcpkg_execute_required_process)
  ports/matplotplusplus/portfile.cmake:34 (vcpkg_cmake_configure)
  scripts/ports.cmake:147 (include)
```
**E:\matplotplusplus\buildtrees\matplotplusplus\config-x86-windows-out.log:**
```
CMake Error: install(EXPORT "Matplot++Targets" ...) includes target "matplot_opengl" which requires target "glad" that is not in any export set.
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    _VCPKG_ROOT_DIR


CMake Generate step failed.  Build files cannot be regenerated correctly.
```

Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review".

**Comment by jiayuehua** [comment]
@monical

> The following error occurred when testing the feature on `x86-windows` and `x64-windows-static`:
> 
> ```
> CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:112 (message):
>     Command failed: "C:/Program Files/Microsoft Visual Studio/2022/Enterprise/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" -v
>     Working Directory: E:/matplotplusplus/buildtrees/matplotplusplus/x86-windows-rel/vcpkg-parallel-configure
>     Error code: 1
>     See logs for more information:
>       E:\matplotplusplus\buildtrees\matplotplusplus\config-x86-windows-dbg-CMakeCache.txt.log
>       E:\matplotplusplus\buildtrees\matplotplusplus\config-x86-windows-rel-CMakeCache.txt.log
>       E:\matplotplusplus\buildtrees\matplotplusplus\config-x86-windows-out.log
> 
> Call Stack (most recent call first):
>   installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_configure.cmake:248 (vcpkg_execute_required_process)
>   ports/matplotplusplus/portfile.cmake:34 (vcpkg_cmake_configure)
>   scripts/ports.cmake:147 (include)
> ```
> 
> **E:\matplotplusplus\buildtrees\matplotplusplus\config-x86-windows-out.log:**
> 
> ```
> CMake Error: install(EXPORT "Matplot++Targets" ...) includes target "matplot_opengl" which requires target "glad" that is not in any export set.
> -- Generating done
> CMake Warning:
>   Manually-specified variables were not used by the project:
> 
>     _VCPKG_ROOT_DIR
> 
> 
> CMake Generate step failed.  Build files cannot be regenerated correctly.
> ```
> 
> Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review".

@MonicaLiu0311 Does that you are using MATPLOTPP_BUILD_EXPERIMENTAL_OPENGL_BACKEND feature?

**Comment by MonicaLiu0311** [comment]
> @MonicaLiu0311 Does that you are using MATPLOTPP_BUILD_EXPERIMENTAL_OPENGL_BACKEND feature?

I use the command `./vcpkg install matplotplusplus[*]:x86-windows` to test all features.

**Comment by jiayuehua** [comment]
will you please test again?

**Comment by dg0yt** [comment]
Duplicates #31229.

**Comment by jiayuehua** [comment]
> Duplicates #31229.

yeah, #31229 have not test feature yet, I hope #31229 can fix opengl feature . It's fine to close this pr.

**Comment by MonicaLiu0311** [comment]
All features are tested successfully in the following triplet:
```
x86-windows
x64-windows
x64-windows-static
```

**Comment by MonicaLiu0311** [comment]
usage:
```
matplotplusplus provides CMake targets:

    # this is heuristically generated, and may not be correct
    find_package(Matplot++ CONFIG REQUIRED)
    # note: 1 additional targets are not displayed.
    target_link_libraries(main PRIVATE Matplot++::cimg std::filesystem Matplot++::matplot Matplot++::nodesoup)
```

When testing usage, the following error occurs:
```
1> [CMake] CMake Error at CMakeFindUsage/CMakeLists.txt:15 (target_link_libraries):
1> [CMake]   Target "main" links to:
1> [CMake] 
1> [CMake]     std::filesystem
1> [CMake] 
1> [CMake]   but the target was not found.  Possible reasons include:
1> [CMake] 
1> [CMake]     * There is a typo in the target name.
1> [CMake]     * A find_package call is missing for an IMPORTED target.
1> [CMake]     * An ALIAS target is missing.
1> [CMake] 
1> [CMake] 
1> [CMake] 
1> [CMake] CMake Error at E:/matplotplusplus/installed/x64-windows/share/matplot++/Matplot++Targets.cmake:68 (set_target_properties):
1> [CMake]   The link interface of target "Matplot++::cimg" contains:
1> [CMake] 
1> [CMake]     ZLIB::ZLIB
1> [CMake] 
1> [CMake]   but the target was not found.  Possible reasons include:
1> [CMake] 
1> [CMake]     * There is a typo in the target name.
1> [CMake]     * A find_package call is missing for an IMPORTED target.
1> [CMake]     * An ALIAS target is missing.
1> [CMake] 
1> [CMake] Call Stack (most recent call first):
1> [CMake]   E:/matplotplusplus/installed/x64-windows/share/matplot++/Matplot++Config.cmake:50 (include)
1> [CMake]   E:/matplotplusplus/scripts/buildsystems/vcpkg.cmake:853 (_find_package)
1> [CMake]   CMakeFindUsage/CMakeLists.txt:14 (find_package)
1> [CMake] -- Generating done (0.1s)
1> [CMake] CMake Generate step failed.  Build files cannot be regenerated correctly.
```

<details>
<summary><b>CMakeFindUsage.cpp</b></summary>
<pre>
#include &#60;iostream&#62;
#include "matplot/matplot.h"

using namespace std;

int main()
{
        cout << "Hello CMake." << endl;
        return 0;
}
</pre></details>


<details>
<summary><b>CMakeLists.txt</b></summary>
<pre>
cmake_minimum_required (VERSION 3.8)

set(CMAKE_TOOLCHAIN_FILE "E:/matplotplusplus/scripts/buildsystems/vcpkg.cmake")

project ("CMakeFindUsage")

add_executable (main "CMakeFindUsage.cpp")

find_package(Matplot++ CONFIG REQUIRED)
target_link_libraries(main PRIVATE Matplot++::cimg std::filesystem Matplot++::matplot Matplot++::nodesoup)
</pre></details>

**Comment by jiayuehua** [comment]
too complex, I close it.

**Comment by jiayuehua** [comment]
should be ok now, @MonicaLiu0311

**Comment by jiayuehua** [comment]
should be ok now, I have test it.

**Comment by Nosenzor** [comment]
Merge ! merge ! merge! ;-)

**Comment by MonicaLiu0311** [comment]
The usage test passed (header files found):
```
The package matplotplusplus provides CMake targets:

    find_package(Matplot++ CONFIG REQUIRED)
    target_link_libraries(main PRIVATE Matplot++::cimg  Matplot++::matplot)
```

File: ports/nodesoup/fix-cmakelists.patch [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+index dbd43b9..2506587 100644
+--- a/src/CMakeLists.txt
++++ b/src/CMakeLists.txt
+@@ -5,7 +5,9 @@ add_library(nodesoup::nodesoup ALIAS nodesoup)
+ target_include_directories(
+   nodesoup
+   PUBLIC $<BUILD_INTERFACE:${nodesoup_SOURCE_DIR}/include>
+-  PRIVATE $<BUILD_INTERFACE:${nodesoup_SOURCE_DIR}/src>)
++  PRIVATE $<BUILD_INTERFACE:${nodesoup_SOURCE_DIR}/src>
++  INTERFACE $<INSTALL_INTERFACE:include>
```
Comment (dg0yt) [inline ports/nodesoup/fix-cmakelists.patch:83]: Nitpick: IMO the fix (for upstream) should be:
~~~diff
  PUBLIC $<BUILD_INTERFACE:${nodesoup_SOURCE_DIR}/include>
+        $<INSTALL_INTERFACE:include>
   PRIVATE $<BUILD_INTERFACE:${nodesoup_SOURCE_DIR}/src>)

File: ports/nodesoup/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,23 @@
+vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
+
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO olvb/nodesoup
+    REF 3158ad0
+    SHA512 be98cd5a1106fb1b6e6cb6b880229f590c2d4c4cc176dcceb2e2226ff3f2344ccb4510fb3a0911e9329701af50f076ee2efb9a3afc9e985b4d9c3fb92c12102d
+    HEAD_REF main
```
Comment (MonicaLiu0311) [inline ports/nodesoup/portfile.cmake:8]:
```suggestion
    HEAD_REF master
```
Be consistent with upstream.

File: ports/nodesoup/fix-cmakelists.patch [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+-  set(CPACK_SOURCE_GENERATOR "TGZ;ZIP")
+-  set(CPACK_SOURCE_IGNORE_FILES /.clang-format /compile_commands.json /.git
+-                                /dist /.*build.* /\\\\.DS_Store)
+-  include(CPack)
+-endif()
+-
+-if((CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME OR nodesoup_BUILD_DEMO)
+-   AND BUILD_DEMO)
+-  add_subdirectory(demo)
+-endif()
```
Comment (dg0yt) [inline ports/nodesoup/fix-cmakelists.patch:72]: This chunk isn't needed.

File: ports/nodesoup/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+)
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
+    OPTIONS 
+    -DBUILD_DEMO=OFF
+)
+vcpkg_cmake_install()
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
+#vcpkg_cmake_config_fixup(PACKAGE_NAME nodesoup CONFIG_PATH lib/cmake/nodesoup)
```
Comment (MonicaLiu0311) [inline ports/nodesoup/portfile.cmake:20]: Remove useless comments.

Comment (MonicaLiu0311) [inline ports/nodesoup/portfile.cmake:20]:
```suggestion
```

File: ports/nodesoup/fix-cmakelists.patch [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+-  COMPATIBILITY AnyNewerVersion)
+ 
+ install(
+   EXPORT nodesoup-targets
+-  FILE nodesoup-config.cmake
++  DESTINATION share/nodesoup
+   NAMESPACE nodesoup::
+-  DESTINATION lib/cmake/nodesoup)
++  FILE nodesoup-config.cmake
++  )
```
Comment (dg0yt) [inline ports/nodesoup/fix-cmakelists.patch:52]: This change isn't needed. It can be handed by `vcpkg_cmake_config_fixup(.. CONFIG_PATH lib/cmake/nodesoup)`.

File: ports/nodesoup/fix-cmakelists.patch [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+-  INCLUDES
+-  DESTINATION include
+-  PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/nodesoup)
++  PUBLIC_HEADER DESTINATION include)
+ 
+-include(CMakePackageConfigHelpers)
+-write_basic_package_version_file(
+-  nodesoup-config-version.cmake
+-  VERSION ${PACKAGE_VERSION}
+-  COMPATIBILITY AnyNewerVersion)
```
Comment (dg0yt) [inline ports/nodesoup/fix-cmakelists.patch:43]: This change is not needed.

File: ports/nodesoup/usage [hunk 7]
```diff
@@ -0,0 +1,4 @@
+The package s2geometry provides CMake targets:
```
Comment (MonicaLiu0311) [inline ports/nodesoup/usage:1]:
```suggestion
The package nodesoup provides CMake targets:
```

File: ports/nodesoup/fix-cmakelists.patch [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+-include(GNUInstallDirs)
+ install(
+   TARGETS nodesoup
+   EXPORT nodesoup-targets
+   LIBRARY DESTINATION lib
+   ARCHIVE DESTINATION lib
+-  INCLUDES
+-  DESTINATION include
+-  PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/nodesoup)
++  PUBLIC_HEADER DESTINATION include)
```
Comment (dg0yt) [inline ports/nodesoup/fix-cmakelists.patch:37]: Why?

File: ports/nodesoup/fix-cmakelists.patch [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+-if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
+-  message(
+-    STATUS
+-      "Setting build type to '${default_build_type}' as none was specified.")
+-  set(CMAKE_BUILD_TYPE
+-      "${default_build_type}"
+-      CACHE STRING "Choose the type of build." FORCE)
+-  set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release"
+-                                               "MinSizeRel" "RelWithDebInfo")
+-endif()
```
Comment (dg0yt) [inline ports/nodesoup/fix-cmakelists.patch:24]: This change is not needed.

File: ports/nodesoup/vcpkg.json [hunk 10]
```diff
@@ -0,0 +1,16 @@
+{
+  "name": "nodesoup",
+  "version": "0.1.0",
```
Comment (MonicaLiu0311) [inline ports/nodesoup/vcpkg.json:3]:
```suggestion
  "version-date": "2023-06-12",
```
The upstream does not have any Releases and Tags. In this case, `version-date` is recommended.

File: ports/matplotplusplus/fix-dependencies.patch [hunk 11]
```diff
@@ -94,3 +171,111 @@ index ab58bbd..f9fed7e 100644
    endif()
  else()
    message("No CMAKE_MODULE_PATH path for OpenCV configured")
+@@ -153,4 +169,4 @@ endif()
+ if(MASTER_PROJECT AND NOT BUILD_SHARED_LIBS)
+   install(TARGETS cimg
+       EXPORT Matplot++Targets)
+-endif()
+\ No newline at end of file
++endif()
```
Comment (dg0yt) [inline ports/matplotplusplus/fix-dependencies.patch:180]: Chunk not needed.

File: ports/matplotplusplus/fix-dependencies.patch [hunk 12]
```diff
@@ -94,3 +171,111 @@ index ab58bbd..f9fed7e 100644
...
+diff --git a/source/CMakeLists.txt b/source/CMakeLists.txt
+index f66dbb4..cf7a6d2 100644
+--- a/source/CMakeLists.txt
++++ b/source/CMakeLists.txt
+@@ -1,2 +1,2 @@
+ add_subdirectory(3rd_party)
+-add_subdirectory(matplot)
+\ No newline at end of file
++add_subdirectory(matplot)
```
Comment (dg0yt) [inline ports/matplotplusplus/fix-dependencies.patch:189]: Chunk not needed.

File: ports/matplotplusplus/fix-dependencies.patch [hunk 13]
```diff
@@ -94,3 +171,111 @@ index ab58bbd..f9fed7e 100644
...
+diff --git a/source/matplot/CMakeLists.txt b/source/matplot/CMakeLists.txt
+index fe9ad53..af3ef72 100644
+--- a/source/matplot/CMakeLists.txt
++++ b/source/matplot/CMakeLists.txt
+@@ -1,5 +1,4 @@
+ if(NOT MINGW)
+-find_package(Filesystem REQUIRED COMPONENTS Experimental Final)
```
Comment (dg0yt) [inline ports/matplotplusplus/fix-dependencies.patch:196]: Hm, do we have to remove support for the experimental implemetations of std::filesystem?

File: ports/matplotplusplus/fix-dependencies.patch [hunk 14]
```diff
@@ -94,3 +171,111 @@ index ab58bbd..f9fed7e 100644
...
+     if (CXX_FILESYSTEM_IS_EXPERIMENTAL)
+-      target_compile_definitions(matplot PRIVATE CXX_FILESYSTEM_IS_EXPERIMENTAL)
+     endif()
+ endif()
+ # Some hack to not depend on FILE* internals
+@@ -214,32 +213,12 @@ if (MATPLOTPP_BUILD_EXPERIMENTAL_OPENGL_BACKEND)
+     # The biggest con of the OpenGL backend is that it cannot open a window
+     #     in another thread. All it can do is get in the middle of the render
+     #     loop and draw the plot.
+-    find_package(OpenGL REQUIRED)
```
Comment (dg0yt) [inline ports/matplotplusplus/fix-dependencies.patch:230]: Is this removal necessary?

File: ports/matplotplusplus/fix-dependencies.patch [hunk 15]
```diff
@@ -94,3 +171,111 @@ index ab58bbd..f9fed7e 100644
...
+-        FetchContent_MakeAvailable(glad)
+-    else ()
+-        # FindGLAD does not usually create a target, so we create an interface target
+-        if (NOT TARGET glad)
+-            add_library(glad INTERFACE)
+-            target_include_directories(glad INTERFACE ${GLAD_INCLUDE_PATH})
+-            target_link_libraries(glad INTERFACE ${GLAD_LIBRARIES})
+-        endif ()
+-    endif ()
++    find_package(glad CONFIG REQUIRED)
```
Comment (dg0yt) [inline ports/matplotplusplus/fix-dependencies.patch:248]: For a shorter patch, only change the first line to:
~~~cmake
find_package(GLAD NAMES glad REQUIRED)
add_target(glad ALIAS glad::glad)
~~~
This will set `GLAD_FOUND` and add the `glad` target as desired.

File: ports/matplotplusplus/fix-dependencies.patch [hunk 16]
```diff
@@ -94,3 +171,111 @@ index ab58bbd..f9fed7e 100644
...
+index b22b7b1..4633dc4 100644
+--- a/source/matplot/axes_objects/network.cpp
++++ b/source/matplot/axes_objects/network.cpp
+@@ -732,4 +732,4 @@ namespace matplot {
+         return marker_colors_;
+     }
+ 
+-} // namespace matplot
+\ No newline at end of file
++} // namespace matplot
```
Comment (dg0yt) [inline ports/matplotplusplus/fix-dependencies.patch:281]: This chunk is not needed.

File: ports/matplotplusplus/fix-dependencies.patch [hunk 17]
```diff
@@ -1,8 +1,85 @@
...
+ @PACKAGE_INIT@
++include(CMakeFindDependencyMacro)
+ 
+ # How this Matplot++ installation was built
+ set(MATPLOT_BUILT_SHARED "@MATPLOTPP_BUILD_SHARED_LIBS@")
+@@ -10,12 +11,44 @@ if (NOT CMAKE_CXX_COMPILER_ID STREQUAL MATPLOT_BUILT_CXX_COMPILER_ID)
+     message(WARNING "This installation of Matplot++ was built with ${MATPLOT_BUILT_CXX_COMPILER_ID}.")
+ endif()
+ 
++if (@MATPLOTPP_BUILD_EXPERIMENTAL_OPENGL_BACKEND@)
```
Comment (dg0yt) [inline ports/matplotplusplus/fix-dependencies.patch:15]: I would recommend to add double quotes to `@VAR@` in `if` conditions. It prevents some unexpected interpretations when evaluated.

File: ports/matplotplusplus/portfile.cmake [hunk 18]
```diff
@@ -9,18 +9,17 @@ vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
...
     REPO alandefreitas/matplotplusplus
-    REF 36d8dc6c3b94b7a71c4f129763f2c6ad8fc0b54a
-    SHA512 ac8902e953a2a9f6bd62e14e2eb0bd42e407bae6c0b2921ad16ce547e4921ba2c8d8a9cc68e75831676dce3cd89cdf8294862710e838510b68e20f8a6cdf806f
+    REF "v${VERSION}"
+    SHA512 9193381fd9d4925259f28a03da33231b3da1273237e7510b93da1ca076610455746feb4ea0d8aa9a4a9e46a5f5f344fc1322f7ffc57963837dbe8b0be0b811ac
     HEAD_REF master
     PATCHES
-        install-3rd-libraries.patch # Remove this patch when nodesoup is added in vcpkg
-        fix-dependencies.patch
+            fix-dependencies.patch
```
Comment (dg0yt) [inline ports/matplotplusplus/portfile.cmake:16]: `install-3rd-libraries.patch` isn't removed from git.
`fix-opengl-feature.patch` isn't used here.

**Review by dg0yt — CHANGES_REQUESTED** [review]
This PR needs a cleanup.

File: ports/matplotplusplus/fix-dependencies.patch (hunk 13 above)
Comment (jiayuehua) [reply ports/matplotplusplus/fix-dependencies.patch:176]: otherwise os can't build successfully.

**Comment by MonicaLiu0311** [comment]
```
##[error]Found the following errors:
##[error]versions/n-/nodesoup.json(2,): error : while parsing versions for nodesoup from version/n-/nodesoup.json
2023-06-12 is declared with c6b844a6292ac5a4c45439d2e9bd3755df98f98f, but the local port has a different SHA 43d30f8e5e0cd4b45a590027d0c9ea5036f0ba40.
```

There is a CI error, please push after running the command `./vcpkg x-add-version nodesoup --overwrite-version`.

`--overwrite-version`
After the first modification, please run `./vcpkg x-add-version portName` to generate the corresponding json; for subsequent modifications of the same version, please use `./vcpkg x-add-version portName --overwrite-version` to update git-tree.

**Comment by jiayuehua** [comment]
@MonicaLiu0311  now this is is ready  .

**Review by MonicaLiu0311 — APPROVED** [review]

File: ports/matplotplusplus/fix-dependencies.patch [hunk 19]
```diff
@@ -94,3 +167,66 @@ index ab58bbd..f9fed7e 100644
...
+-      target_compile_definitions(matplot PRIVATE CXX_FILESYSTEM_IS_EXPERIMENTAL)
+     endif()
+ endif()
+ # Some hack to not depend on FILE* internals
+@@ -217,7 +216,7 @@ if (MATPLOTPP_BUILD_EXPERIMENTAL_OPENGL_BACKEND)
+     find_package(OpenGL REQUIRED)
+ 
+     # https://github.com/Dav1dde/glad
+-    find_package(GLAD QUIET)
++    find_package(GLAD name glad CONFIG REQUIRED)
```
Comment (autoantwort) [inline ports/matplotplusplus/fix-dependencies.patch:211]: There is no "name" parameter in `find_package`

Comment (MonicaLiu0311) [reply ports/matplotplusplus/fix-dependencies.patch:211]: Will amend as soon as possible.
